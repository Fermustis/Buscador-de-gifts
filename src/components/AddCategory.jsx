import { useState } from "react"
import PropTypes from 'prop-types';


export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = (e) => {
        setInputValue(e.target.value)
    }
    const onSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length <= 1) return;

        // setCategories(cats => [inputValue, ...cats]);

        onNewCategory(inputValue.trim());

        setInputValue('');

    }
    return (
        <>
            <form onSubmit={(e) => onSubmit(e)} aria-label="form">

                {/* Input */}
                <input className="mb-6 bg-gray-100 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 cursor-not-allowed dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-gray-400 dark:focus:ring-blue-500 dark:focus:border-blue-500" type="text" placeholder="Buscar gift" value={inputValue} onChange={onInputChange} />

                <hr />
            </form>
        </>
    )
}
AddCategory
    .propTypes = {
    onNewCategory: PropTypes.func.isRequired
}