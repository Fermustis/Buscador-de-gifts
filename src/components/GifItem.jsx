import PropTypes from 'prop-types';

export const GifItem = ({ title, url }) => {

    return (
        <div className="max-w-sm bg-white border border-gray-200 rounded-lg shadow dark:bg-gray-800 dark:border-gray-70">

            <img className="h-auto max-w-full rounded-lg" src={url} alt={title} />
            <p className='mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white'>{title}</p>
        </div>
    )
}
GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}