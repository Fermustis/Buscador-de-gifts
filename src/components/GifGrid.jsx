import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import Proptypes from 'prop-types';



export const GifGrid = ({ category }) => {


    const { images, loading } = useFetchGifs(category);


    return (
        <>
            <h3>{category}</h3>
            {loading && <p>Loading...</p>}
            <div className='grid grid-cols-2 md:grid-cols-3 gap-4'>
                {
                    images.map((image) => (

                        <GifItem key={image.id}
                            {...image}
                        />
                    ))

                }
            </div>


        </>

    )

};
GifGrid.propTypes = {
    category: Proptypes.string.isRequired
}