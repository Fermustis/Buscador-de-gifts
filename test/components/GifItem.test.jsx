import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas <GifItem/>', () => {

    const title = 'Mustis';
    const url = 'https://localhost/algo.jpg';
    test('Debe mostrar el componente correctamente ', () => {


        const { container } = render(<GifItem title={title} url={url} />);

        expect(container).toMatchSnapshot();
    });

    test('Debe mostra la imagen con URL y el  ALT indicado', () => {

        render(<GifItem title={title} url={url} />);
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        //otra manera mas elegante de hacerlo

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);

    });
    test('debe tener el titulo en el componente', () => {

        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();// toBeTruthy() es para que exista
    });



});