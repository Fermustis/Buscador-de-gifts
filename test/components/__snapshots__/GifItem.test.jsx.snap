// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifItem/> Debe mostrar el componente correctamente  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Mustis"
      src="https://localhost/algo.jpg"
    />
    <p>
      Mustis
    </p>
  </div>
</div>
`;
